src/components/ApiDocumentation/ApiTypeIndicator.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { style } from 'typestyle';
+
+interface Props {
+  apiType: string;
+}
+
+const apiGrpcIcon = require('../../assets/img/api-grpc-logo.svg');
+const apiRestIcon = require('../../assets/img/api-rest-logo.svg');
+const apiGraphqlIcon = require('../../assets/img/api-graphql-logo.svg');
+
+const nameToSource = new Map<string, string>([
+  ['grpc', apiGrpcIcon],
+  ['rest', apiRestIcon],
+  ['graphql', apiGraphqlIcon],
+]);
+
+export class ApiTypeIndicator extends React.Component<Props> {
+  iconStyle() {
+    return style({
+      marginTop: -2,
+      marginRight: 6,
+      width: 30
+    });
+  }
+
+  render() {
+    if (this.props.apiType) {
+      return this.renderIcon(this.props.apiType);
+    }
+    return <span />;
+  }
+
+  renderIcon(apiType: string) {
+    if (this.props.apiType === 'grpc' || this.props.apiType === 'graphql' || this.props.apiType === 'rest') {
```

Comment:
This is somewhat better, and given that we have the `nameToSource` to see if there's an icon, maybe we should check if the key is there instead of enumerating all the possibilities here too?

Reply:
Good point, I fixed it